server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -440,6 +441,23 @@ public ShardSnapshotStatus status() {
         public String toString() {
             return snapshot + ", shardId [" + shardId + "], status [" + status.state() + "]";
         }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            final UpdateIndexShardSnapshotStatusRequest that = (UpdateIndexShardSnapshotStatusRequest) o;
+            return snapshot.equals(that.snapshot) && shardId.equals(that.shardId) && status.equals(that.status);
```

Comment:
nit: I think all fields of this class can be final now. Can we also add a test using EqualsHashCodeTestUtils?

Reply:
Thanks Nhat! How about the tests in e21440d4c5b  ? :)